src/model/names.ml

```diff
@@ -1,19 +1,22 @@
-let parenthesise name =
-  match name with
-  | "asr" | "land" | "lor" | "lsl" | "lsr" | "lxor" | "mod" -> "(" ^ name ^ ")"
-  | _ ->
-      if String.length name > 0 then
-        match name.[0] with
-        | 'a' .. 'z'
-        | '\223' .. '\246'
-        | '\248' .. '\255'
-        | '_'
-        | 'A' .. 'Z'
-        | '\192' .. '\214'
-        | '\216' .. '\222' ->
-            name
-        | _ -> "(" ^ name ^ ")"
-      else name
+open Odoc_utils
+
+(** Returns [true] on chars that are part of operators. *)
+let operator_char = function
+  (* https://ocaml.org/manual/5.2/lex.html#core-operator-char *)
+  | '$' | '&' | '*' | '+' | '-' | '/' | '=' | '>' | '@' | '^' | '|' | '~' | '!'
+  | '?' | '%' | '<' | ':' | '.'
+  (* https://ocaml.org/manual/5.2/lex.html#infix-symbol *)
+  | '#'
+  (* https://ocaml.org/manual/5.2/indexops.html#s:index-operators *)
+  | '(' | ')' | '[' | ']' | '{' | '}' ->
+      true
+  | _ -> false
+
+let is_operator = function
+  | "asr" | "land" | "lor" | "lsl" | "lsr" | "lxor" | "mod" | "or" -> true
+  | name -> String.exists operator_char name
+
+let parenthesise name = if is_operator name then "(" ^ name ^ ")" else name
```

Comment:
I would suggest to do `"( " ^ name ^ " )"` because of the ambiguous `(*)`.

Reply:
That's a good point!

Changing this code would change URLs, which is not fine. More work is needed to handle that in the generated code only. It's in `src/document/generator.ml` if you have some time to work on it.